src/nvim/screen.c

```diff
@@ -4456,14 +4465,24 @@ static int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool noc
   return row;
 }
 
-void draw_virt_text(buf_T *buf, int col_off, int *end_col, int max_col)
+void win_extmarks_add(uint64_t ns_id, uint64_t mark_id, int start_row, int end_row, int win_col) {
+  Array m = ARRAY_DICT_INIT;
+  ADD(m, INTEGER_OBJ(ns_id));
+  ADD(m, INTEGER_OBJ(mark_id));
+  ADD(m, INTEGER_OBJ(start_row));
+  ADD(m, INTEGER_OBJ(end_row));
+  ADD(m, INTEGER_OBJ(win_col));
```

Comment:
supplying both start and end row but only one column is a little odd. An alternative would be to pass in `int win_row` (`row` var in `win_line() ) also to `decor_redraw_col`. Then it could handle grid (row,col) positions for both ends of an extrange, and avoid the extra check in `draw_virt_text`.

Reply:
I'm still not getting this right -- the mark is not sent when the start row is above the viewport but end row is in.

Reply:
in that case, let's remove `end_row`. we can add `end_row` / `end_col` pair later.

Reply:
roger roger

Reply:
done.